utils.groovy

```diff
@@ -933,19 +933,54 @@ def build_and_push_image(params = [:]) {
     // src_url:              string  -- Source Git URL.
     // staging_repository:   string  -- Repository URL for the staging repo.
     // v2s2:                 bool    -- Use docker v2s2 format rather than OCI. Default to false.
+    // check_base_match      bool    -- Wether to check if the base image is the same accross all arches. Default to false.
 
     def secret = params.get('secret', "");
     def from = params.get('from', "");
     def manifest_digest = ""
     def extra_build_args = params.get('extra_build_args', "");
     def v2s2 = params.get('v2s2', false);
+    def arches = params['arches']
+    def check_base_match = params.get('check_base_match', false)
 
-    def digests = build_remote_image(params['arches'], params['src_commit'], params['src_url'], params['staging_repository'],
+    def from_manifest = skopeo_inspect(from)
+    if (!from.contains('@')) {
+        // pin the FROM with the manifest digest, to avoid arch remote builders reusing stale images
+        from = "${from.split(':')[0]}@${from_manifest["Digest"]}"
```

Comment:
This will be an arch-specific digest. We need the manifest list digest.

Which actually... I don't see a way to get that from skopeo.

OK, this seems to work:

```
$ skopeo inspect --raw -n docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev:4.19-9.6-node-image > manifest.json
$ skopeo manifest-digest manifest.json
sha256:5fbb06ed936e20174a301c9c4e2dcf17a2e739fa223630440c2868cb7f45c1c5
```

Let's drop the `skopeo_inspect` function and replace it with a `get_manifest_digest()`?

Reply:
OK sorry, I was wrong here. The `.Digest` field of a `skopeo inspect docker://manifest/listed/image` _is_ the manifest list digest.

One interesting semi-gotcha though is that `--override-arch` doesn't affect the reported digest at all (i.e. it's still the digest of the manifest list), but the manifests are clearly different. Which I guess makes sense but is a bit surprising.

Reply:
This is https://github.com/containers/skopeo/issues/1554 I think